src/Services/AppInstallActivationHandler.cs

```diff
@@ -51,63 +63,107 @@ protected override bool CanHandleInternal(ProtocolActivatedEventArgs args)
 
     protected async override Task HandleInternalAsync(ProtocolActivatedEventArgs args)
     {
-        await AppActivationFlowAsync(args.Uri.Query);
-    }
+        var uri = args.Uri;
+        var parameters = HttpUtility.ParseQueryString(uri.Query);
 
-    private async Task AppActivationFlowAsync(string query)
-    {
-        try
+        if (parameters != null)
         {
-            // Don't interrupt the user if the machine configuration is in progress
-            if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+            var searchQuery = parameters.Get("search");
+            var wingetIdsQuery = parameters.Get("wingetIds");
+
+            if (!string.IsNullOrEmpty(searchQuery))
             {
-                _log.Warning("Cannot activate the add-apps-to-cart flow because the machine configuration is in progress");
-                return;
+                await AppActivationFlowAsync(searchQuery, ActivationQueryType.Search);
             }
-            else
+            else if (!string.IsNullOrEmpty(wingetIdsQuery))
             {
-                _log.Information("Starting add-apps-to-cart activation");
-                _navigationService.NavigateTo(typeof(SetupFlowViewModel).FullName!);
-                _setupFlowViewModel.StartAppManagementFlow(query);
-                await SearchAndSelectAsync(query);
+                await AppActivationFlowAsync(wingetIdsQuery, ActivationQueryType.WingetIds);
             }
         }
-        catch (Exception ex)
-        {
-            _log.Error(ex, "Error executing the add-apps-to-cart activation flow");
-        }
     }
 
-    private async Task SearchAndSelectAsync(string query)
+    private async Task AppActivationFlowAsync(string query, ActivationQueryType queryType)
     {
-        var parameters = HttpUtility.ParseQueryString(query);
-        var searchParameter = parameters["search"];
+        if (_setupFlowOrchestrator.IsMachineConfigurationInProgress)
+        {
+            _log.Warning("Cannot activate the add-apps-to-cart flow because the machine configuration is in progress");
+            await _mainWindow.ShowErrorMessageDialogAsync(
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.AppInstallActivationTitle),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.URIActivationFailedBusy),
+                    _setupFlowStringResource.GetLocalized(StringResourceKey.Close));
+            return;
+        }
 
-        if (string.IsNullOrEmpty(searchParameter))
+        var identifiers = ParseIdentifiers(query, queryType);
```

Comment:
Suggest to turn this into a List instead of an Array.  They wat you can remove all the `.Length == 0` checks.

Reply:
wouldn't those just flip to `.Count == 0` instead or is there a better way?

Reply:
... I totally thought I saw places that would benefit from a List.  I guess I was mistaken.  My bad.